gene2phenotype_project/gene2phenotype_app/tests/panel.py

```diff
@@ -0,0 +1,74 @@
+from django.test import TestCase
+from django.urls import reverse
+
+
+class PanelListEndpointTests(TestCase):
+    """
+        Test the panel endpoint: PanelList
+    """
+    fixtures = ["gene2phenotype_app/fixtures/user_panels.json", "gene2phenotype_app/fixtures/attribs.json"]
+
+    def setUp(self):
+        self.url_panels = reverse('list_panels')
+
+    def test_get_panel_list(self):
+        """
+            Test for non-autenticated users.
+            Non-visible panels are not included in the response.
+        """
+        response = self.client.get(self.url_panels)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data.get("count"), 2)
+
+    def test_login_get_panel_list(self):
+        """
+            Test for autenticated users.
+            All panels are included in the response.
+        """
+        login = self.client.login(username="user5@test.ac.uk", password="user5pass")
+        self.assertTrue(login)
+
+        response = self.client.get(self.url_panels)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data.get("count"), 3)
+
+class PanelDetailsEndpointTests(TestCase):
+    """
+        Test the panel endpoint: PanelDetail
+    """
+    fixtures = ["gene2phenotype_app/fixtures/user_panels.json", "gene2phenotype_app/fixtures/attribs.json",
+                "gene2phenotype_app/fixtures/g2p_stable_id.json", "gene2phenotype_app/fixtures/locus.json",
+                "gene2phenotype_app/fixtures/sequence.json", "gene2phenotype_app/fixtures/disease.json",
+                "gene2phenotype_app/fixtures/ontology_term.json", "gene2phenotype_app/fixtures/source.json",
+                "gene2phenotype_app/fixtures/locus_genotype_disease.json", "gene2phenotype_app/fixtures/lgd_panel.json"]
+
+    def setUp(self):
+        self.url_panels = reverse('panel_details', kwargs={'name': 'DD'})
+
+    def test_get_panel_details(self):
+        """
+            Get the details for a visible panel.
+        """
+        response = self.client.get(self.url_panels)
+        self.assertEqual(response.status_code, 200)
```

Comment:
Suggesting we should add another test here, that checks for a panelDetail maybe description

Reply:
self.assertEqual(response.data.get('description'), 'Developmental disorders') - we can add this @dglemos

Reply:
Test added.